src/damavand/cloud/aws/resources/vllm_component.py

```diff
@@ -359,41 +354,226 @@ def api_resource_completions(self) -> aws.apigateway.Resource:
+        """
+        if not self.args.api_key_required:
+            raise AttributeError("`admin_api_secret` is only available when api_key_required is False")
+
+        return aws.secretsmanager.Secret(
+            resource_name=f"{self._name}-api-key-secret",
+            opts=ResourceOptions(parent=self),
+        )
+
+    @property
+    @cache
+    def api_key_secret_version(self) -> aws.secretsmanager.SecretVersion:
+        """
+        Return the secret version for the API key
+
+        Raises
+        ------
+        AttributeError
+            When api_key_required is False.
+        """
+        if not self.args.api_key_required:
+            raise AttributeError("`api_key_secret_version` is only available when api_key_required is False")
+
+        return aws.secretsmanager.SecretVersion(
+            resource_name=f"{self._name}-api-key-secret-version",
+            opts=ResourceOptions(parent=self, depends_on=[self.api_key_secret]),
+            secret_id=self.api_key_secret.id,
+            secret_string=self.admin_api_key.id,
+        )
+
+
+    @property
+    @cache
+    def default_usage_plan(self) -> aws.apigateway.UsagePlan:
+        """
+        Return a default usage plan for the API Gateway, that does not limit the usage.
+
+        Raises
+        ------
+        AttributeError
+            When api_key_required is False.
+        """
+        if not self.args.api_key_required:
+            raise AttributeError("`default_usage_plan` is only available when api_key_required is False")
+
+        return aws.apigateway.UsagePlan(
+            resource_name=f"{self._name}-default-api-usage-plan",
+            opts=ResourceOptions(parent=self),
+            api_stages=[
+                aws.apigateway.UsagePlanApiStageArgs(
+                    api_id=self.api.id,
+                    # NOTE: How do we want to deal with API stages vs. AWS environments?
+                    stage=self.args.api_env_name,
+                )
+            ],
+        )
+
+    @property
+    @cache
+    def tier_1_usage_plan(self) -> aws.apigateway.UsagePlan:
+        """
+        Return a tier 1 usage plan for the API Gateway, with the following limits:
+        - requests per minute: 500
+
+        Raises
+        ------
+        AttributeError
+            When api_key_required is False.
+        """
+        if not self.args.api_key_required:
+            raise AttributeError("`default_usage_plan` is only available when api_key_required is False")
+
+        return aws.apigateway.UsagePlan(
+            resource_name=f"{self._name}-tier-1-api-usage-plan",
+            opts=ResourceOptions(parent=self),
+            api_stages=[
+                aws.apigateway.UsagePlanApiStageArgs(
+                    api_id=self.api.id,
+                    stage=self.args.api_env_name,
+                )
+            ],
+            throttle_settings=aws.apigateway.UsagePlanThrottleSettingsArgs(
+                rate_limit=500
```

Comment:
@kkiani I'm only changing the rate limit at the moment. 

It's also possible to define quotas per day/week/month, but I think that's only interesting when you start pricing API's.

Reply:
@bramelfrink, No I think `rate_limit` is enough.